### ISSUE TITLE
[BUG] Critical data leakage : Scaler fitted on full dataset before train/test split

### ISSUE THREAD

**Issue by rsprajapati7** [abhaycs24/CBSOT_SIP_PROJECT-1-#3]
## Bug Description
`StandardScaler` and `KMeans` are both fitted on the **entire** feature matrix `X`
(all rows) **before** `train_test_split` is called. This means test-set statistics
(mean, variance) influence the scaler, which is a classic form of data leakage that
produces artificially optimistic evaluation metrics.
Additionally, the supervised Random Forest model is then trained on **raw, unscaled**
`X_train`, while clustering used **scaled** features — the two branches operate on
inconsistent feature spaces.
## Problematic Code
```python
# BUG — scaler and KMeans fitted on ALL data (including future test rows)
X_scaled = scaler.fit_transform(X)
kmeans_final.fit_predict(X_scaled)
# Split happens AFTER scaling — too late
X_train, X_test, y_train, y_test = train_test_split(X, y, ...)
# Supervised model trains on raw (unscaled) X_train — inconsistency!
rf_tuned.fit(X_train, y_train)
y_pred = best_model.predict(X_test)

##Impact
- Reported accuracy and AUC scores are invalid — they reflect data the model has already seen through the scaler.
- The Random Forest and the K-Means pipeline operate on different feature scales, making results incomparable.
- This is a fundamental ML correctness violation that invalidates all reported results.

**Comment by rsprajapati7** [comment]
Hey @abhaycs24 I would like to work on this issue on behalf of OSCI'26. Please assign it to me and I will start working on it and fix it ASAP

**Comment by chetna123-ch** [comment]
hey @abhaycs24 i would like to work on this issue under OSCI'26 could you please assign

**Comment by D10sanp** [comment]
Hi @abhaycs24 , I would like to work on this issue as part of OSCI'26.

I’ll fix the data leakage in the supervised churn-prediction pipeline, ensure preprocessing is fitted only on the training data, and verify the updated evaluation metrics.

Please assign this issue to me. Thank you!
